planner: update documents about cost model (#11075)
pingcap/docs: system-variables.md

> **Warning:**
>
> Switching the version of the cost model might cause changes to query plans.

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Default value: `2`
- Value options:
    - `1`: enable the Cost Model Version 1.

>     - `2`: enable the [Cost Model Version 2](/cost-model.md#cost-model-version-2), which is introduced in TiDB v6.2.0 and is more accurate than the previous version in internal tests.

Author: ```suggestion
    - `2`: enable the [Cost Model Version 2](/cost-model.md#cost-model-version-2), which is introduced in TiDB v6.2.0 and is more accurate than the version 1 in internal tests.
```

### tidb_cost_model_version <span class="version-mark">New in v6.2.0</span>

> **Note:**
>
> - Since TiDB v6.5.0, the newly created cluster uses Cost Model Version 2 by default. If you upgrade from a TiDB version earlier than v6.5.0 to v6.5.0 or later, the `tidb_cost_model_version` value does not change.
> - Switching the version of the cost model might cause changes to query plans.

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `2`
- Value options:

>     - `1`: enables the Cost Model Version 1, which is used by default in TiDB v6.4.0 and earlier versions.
>     - `2`: enables the [Cost Model Version 2](/cost-model.md#cost-model-version-2), which is introduced in TiDB v6.2.0 and is more accurate than the version 1 in internal tests.

Author: ```suggestion
    - `1`: enables the Cost Model Version 1, which is used by default in TiDB v6.4.0 and earlier versions.
    - `2`: enables the [Cost Model Version 2](/cost-model.md#cost-model-version-2), which is generally available in TiDB v6.5.0 and is more accurate than the version 1 in internal tests.
```

Reviewer: LGTM

> **Warning:**
>
> Switching the version of the cost model might cause changes to query plans.

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Type: Integer
- Default value: `2`
- Value options:

>     - `1`: enable the Cost Model Version 1.

Author: ```suggestion
    - `1`: enable the Cost Model Version 1, which is used by default in TiDB v6.4.0 and earlier versions. 
```

>     - `1`: enable the Cost Model Version 1, which is used by default in TiDB v6.4.0 and earlier versions. 

Reviewer 2: ```suggestion
    - `1`: enables the Cost Model Version 1, which is used by default in TiDB v6.4.0 and earlier versions. 
```

>     - `2`: enable the [Cost Model Version 2](/cost-model.md#cost-model-version-2), which is introduced in TiDB v6.2.0 and is more accurate than the version 1 in internal tests.

Reviewer 2: ```suggestion
    - `2`: enables the [Cost Model Version 2](/cost-model.md#cost-model-version-2), which is introduced in TiDB v6.2.0 and is more accurate than the version 1 in internal tests.
```

> > **Warning:**

Author: ```suggestion
> **Note:**
```
